e2e/websocket_test.go

```diff
@@ -0,0 +1,161 @@
+			config.Premine(account.address, account.balance)
+		}
+	})
+	srv := srvs[0]
+	client := srv.JSONRPC()
+
+	// Convert the default JSONRPC address to a WebSocket one
+	wsURL := "ws" + strings.TrimPrefix(srv.JsonRPCAddr(), "http") + "/ws"
+
+	// Connect to the websocket server
+	ws, _, err := websocket.DefaultDialer.Dial(wsURL, nil)
+	if err != nil {
+		t.Fatalf("Unable to connect to WS: %v", err)
+	}
+	defer ws.Close()
+
+	t.Run("Valid account balance", func(t *testing.T) {
+		requestID := 1
+		request, constructErr := constructWSRequest(
+			requestID,
+			"eth_getBalance",
+			[]string{preminedAccounts[0].address.String(), "latest"},
+		)
+
+		assert.Nilf(t, constructErr, fmt.Sprintf("Unable to construct request: %v", constructErr))
+
+		res, _ := getWSResponse(t, ws, request)
+
+		assert.Equalf(t, res.ID, float64(requestID), "Invalid response ID")
+
+		var balanceHex string
+		if wsError := json.Unmarshal(res.Result, &balanceHex); wsError != nil {
+			t.Fatalf("Unable to unmarshal WS result: %v", wsError)
+		}
+
+		foundBalance, parseError := types.ParseUint256orHex(&balanceHex)
+		if parseError != nil {
+			t.Fatalf(fmt.Sprintf("Unable to parse WS result balance: %v", parseError))
+		}
+
+		if wsError := json.Unmarshal(res.Result, &balanceHex); wsError != nil {
+			t.Fatalf("Unable to parse WS result balance: %v", wsError)
+		}
+
+		preminedAccounts[0].balance.Cmp(foundBalance)
+		assert.Equalf(t, 0, preminedAccounts[0].balance.Cmp(foundBalance), "Balances don't match")
+	})
+
+	t.Run("Valid block number after transfer", func(t *testing.T) {
+		hash, err := client.Eth().SendTransaction(&web3.Transaction{
+			From:     web3.HexToAddress(srv.Config.PremineAccts[0].Addr.String()),
+			To:       (*web3.Address)(&preminedAccounts[1].address),
+			GasPrice: 10000,
+			Gas:      1000000,
+			Value:    big.NewInt(10000),
+			Nonce:    uint64(0),
+		})
+		assert.NoError(t, err)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		_, err = srv.WaitForReceipt(ctx, hash)
+		assert.NoError(t, err)
+
+		requestID := 2
+		request, constructErr := constructWSRequest(
+			requestID,
+			"eth_blockNumber",
+			[]string{},
+		)
+
+		assert.Nilf(t, constructErr, fmt.Sprintf("Unable to construct request: %v", constructErr))
+
+		res, _ := getWSResponse(t, ws, request)
+
+		assert.Equalf(t, res.ID, float64(requestID), "Invalid response ID")
+
+		var blockNum string
+		if wsError := json.Unmarshal(res.Result, &blockNum); wsError != nil {
+			t.Fatalf("Unable to unmarshal WS result: %v", wsError)
+		}
+
+		blockNumInt, parseError := types.ParseUint256orHex(&blockNum)
+		if parseError != nil {
+			t.Fatalf(fmt.Sprintf("Unable to parse WS result balance: %v", parseError))
+		}
+		if wsError := json.Unmarshal(res.Result, &blockNum); wsError != nil {
+			t.Fatalf("Unable to parse WS result balance: %v", wsError)
+		}
+
+		assert.Greaterf(t, blockNumInt.Cmp(big.NewInt(0)), 0, "Invalid block number")
```

Comment:
Can we do this instead, please?
```suggestion
		assert.Equalf(t, blockNumInt.Cmp(big.NewInt(0)), 1, "Invalid block number")
```

Took me a bit until I understood it actually checks `Greaterf` 😅

Reply:
Changing now 👍